### PR TITLE
docs: PDE Tutorial

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR introduces a tutorial for Palette Dev Engine featuring how to deploy applications into a virtual cluster.
💻 [Preview](https://deploy-preview-1091--docs-spectrocloud.netlify.app/devx/apps/deploy-app)
🎫 [DOC-421](https://spectrocloud.atlassian.net/browse/DOC-421)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:773]:
```suggestion
Log in to [Palette](https://console.spectrocloud.com), navigate to the left **Main Menu**, and select **Apps**. Click on the **scenario-1** row, which takes you to the application’s overview page. Once you are on the scenario-1 overview page, click on the exposed URL for the service. A hyperlink for port 8080 is available.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:812]:
```suggestion
- `spectrocloud_application` - `scenario-2` - the application that will be deployed into cluster-2 that uses the `spectrocloud_application_profile.hello-universe-complete` app profile.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:977]:
```suggestion
The database service block uses a different data resource, `data.spectrocloud_pack_simple.postgres_service`, to find the Postgres service. If you review the data resource, you will find a different type, `operator-instance`. The Postgres service uses a Postgres [operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) to manage the database inside the virtual cluster. 
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1093]:
```suggestion
Open the **inputs.tf** file and set the `variable enable-second-scenario"` default value to `true`.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1105]:
```suggestion
Next, issue the command `terraform apply` to deploy the second scenario. Notice how the `-var` flag is included with the token value in the command. 
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1172]:
```suggestion
To learn more about Palette Dev Engine and its capabilities, check out the reference resource below.
```

**Review by ritawatson — DISMISSED** [review]
A clear tutorial.  A couple suggestions:
- Remove 'on' in 'click on': Change 'click on _element_' to 'click _element_'. I suggest this as part of our internal style. 
- While 'Go ahead and ...' is intended to keep the tone friendly, it seems a bit overused in this tutorial and it adds unnecessary verbiage. Maybe keep it in a hand full of places but remove when introducing a list.

Do forward slashes need to be added in image paths now that images are in one folder?

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1296561603" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:317]:
```suggestion
Provide the UI container with the following information.
```

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1127]:
```suggestion
The last `pack {}` block in the app profile resource `spectrocloud_application_profile.hello-universe-complete` is for the UI. Like the API service, environment variables are used to initialize the UI and the reverse proxy. The UI service requires the URL of the API service and the URL of the public-facing load balancer. Palette output variables are used to populate these two environment variables. A Terraform variable will populate the authentication token required for all API requests.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1193]:
```suggestion
Log in to [Palette](https://console.spectrocloud.com) and navigate to the left **Main Menu**, click on **Apps**. Select the **scenario-2** row. When you are on the scenario-2 overview page, click on the exposed URL for the service. A hyperlink for port 8080 and port 3000 is available.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1210]:
```suggestion
The global counter is no longer available. Instead, you have a counter that starts at zero. Each time you click on the center image, the counter is incremented and stored in the Postgres database along with metadata.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1213]:
```suggestion
## Cleanup
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1232]:
```suggestion
If a cluster remains in the delete phase for over 15 minutes, it becomes eligible for **Force Delete**. To trigger a force delete, navigate to the respective cluster’s details page and click on **Settings**. Click on the **Force Delete Cluster** to delete the cluster. Palette will automatically remove clusters stuck in the cluster deletion phase for over 24 hours.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:1247]:
```suggestion
To learn more about Palette Dev Engine and its capabilities, check out the reference resources below.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (ritawatson) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:432]:
```suggestion
If a cluster remains in the delete phase for over 15 minutes, it becomes eligible for **Force Delete**. To trigger a force delete, navigate to the respective cluster’s details page and click on **Settings**. Click on the **Force Delete Cluster** to delete the cluster. Palette will automatically remove clusters stuck in the cluster deletion phase for over 24 hours.
```

**Review by ritawatson — APPROVED** [review]
Nice tone and clear instruction. Just some some fit and finish remarks.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1320903404" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by tiwarisumit3 — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1091#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1320903404" title="Approved">ritawatson</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1091#pullrequestreview-1321046204" title="Approved">tiwarisumit3</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:42]:
```suggestion
![Architecture diagram depicting two virtual clusters](/tutorials/deploy-app/devx_apps_deploy-apps_architecture-diagram.png)
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:590]:
```suggestion
Next, take a look at the **application-profiles.tf** file. The resource `spectrocloud_application_profile.hello-universe-ui` is the resource responsible for creating the app profile for the first scenario. There are several points of interest in this resource that you should be familiar with. Focus on these five key points:
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (karl-cardenas-coding) [inline content/docs/04.5-devx/06-apps/05-deploy-app.md:739]:
```suggestion
The last Terraform resource to review before deploying the application is located in the **application.tf** file. The resource `spectrocloud_application.hello-universe-ui` is what creates the *app*. In Palette, an app combines a virtual cluster and an app profile. When you deploy an app profile into a virtual cluster, you create an app. This resource points to the app profile `spectrocloud_application_profile.hello-universe-ui` and the cluster resource `spectrocloud_virtual_cluster.cluster-1`. The two resources are required to create an app.
```

File: content/docs/04.5-devx/06-apps/05-deploy-app.md
Comment (karl-cardenas-coding) [reply content/docs/04.5-devx/06-apps/05-deploy-app.md:1213]: @ritawatson We have "clean-up" in the tutorial layout guide.

**Pull request merged**
